plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -706,13 +706,142 @@ export class PersonOverrideWriter {
+        overrideDetails: PersonOverrideDetails
+    ): Promise<ProducerRecord[]> {
+        const mergedAt = DateTime.now()
+
+        await this.postgres.query(
+            tx,
+            SQL`
+                INSERT INTO posthog_flatpersonoverride (
+                    team_id,
+                    old_person_id,
+                    override_person_id,
+                    oldest_event,
+                    version
+                ) VALUES (
+                    ${overrideDetails.team_id},
+                    ${overrideDetails.old_person_id},
+                    ${overrideDetails.override_person_id},
+                    ${overrideDetails.oldest_event},
+                    0
+                )
+            `,
+            undefined,
+            'personOverride'
+        )
+
+        const { rows: transitiveUpdates } = await this.postgres.query(
+            tx,
+            SQL`
+                UPDATE
+                    posthog_flatpersonoverride
+                SET
+                    override_person_id = ${overrideDetails.override_person_id},
+                    version = COALESCE(version, 0)::numeric + 1
+                WHERE
+                    team_id = ${overrideDetails.team_id} AND override_person_id = ${overrideDetails.old_person_id}
+                RETURNING
+                    old_person_id,
+                    version,
+                    oldest_event
+            `,
+            undefined,
+            'transitivePersonOverrides'
+        )
+
+        status.debug('🔁', 'person_overrides_updated', { transitiveUpdates })
+
+        const personOverrideMessages: ProducerRecord[] = [
+            {
+                topic: KAFKA_PERSON_OVERRIDE,
+                messages: [
+                    {
+                        value: JSON.stringify({
+                            team_id: overrideDetails.team_id,
+                            old_person_id: overrideDetails.old_person_id,
+                            override_person_id: overrideDetails.override_person_id,
+                            oldest_event: castTimestampOrNow(overrideDetails.oldest_event, TimestampFormat.ClickHouse),
+                            merged_at: castTimestampOrNow(mergedAt, TimestampFormat.ClickHouse),
+                            version: 0,
+                        }),
+                    },
+                    ...transitiveUpdates.map(({ old_person_id, version, oldest_event }) => ({
+                        value: JSON.stringify({
+                            team_id: overrideDetails.team_id,
+                            old_person_id: old_person_id,
+                            override_person_id: overrideDetails.override_person_id,
+                            oldest_event: castTimestampOrNow(oldest_event, TimestampFormat.ClickHouse),
+                            merged_at: castTimestampOrNow(mergedAt, TimestampFormat.ClickHouse),
+                            version: version,
+                        }),
+                    })),
```

Comment:
This could be made to be less duplicative with the non-flat version, but I don't think it's worth the investment given that the non-flat implementations days should be numbered at this point.